File: 167.cpp
#include <vector>
using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int> &numbers, int target) {
    int l = 0, r = numbers.size() - 1;
    for (; l < r;) {
      int sum = numbers[l] + numbers[r];
      if (sum == target) {
        return {l + 1, r + 1};
      }
      sum > target ? --r : ++l;
    }
    return {l + 1, r + 1};
  }
};

File: 167.py
class Solution:
    def twoSum(self, numbers: list[int], target: int) -> list[int]:
        l, r = 0, len(numbers) - 1
        while l < r:
            summ = numbers[l] + numbers[r]
            if summ == target:
                return [l + 1, r + 1]
            elif summ > target:
                r -= 1
            else:
                l += 1
        return [l + 1, r + 1]
